src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -1,93 +1,213 @@
-            var syntaxFacts = GetSyntaxFactsService();
+            switch (node.Kind())
+            {
+                case SyntaxKind.SwitchSection:
+                    return (int)SyntaxKind.BreakStatement;
+
+                case SyntaxKind.LocalFunctionStatement:
+                case SyntaxKind.SetAccessorDeclaration:
+                case SyntaxKind.GetAccessorDeclaration:
+                case SyntaxKind.AddAccessorDeclaration:
+                case SyntaxKind.RemoveAccessorDeclaration:
+                case SyntaxKind.MethodDeclaration:
+                case SyntaxKind.ConstructorDeclaration:
+                case SyntaxKind.DestructorDeclaration:
+                case SyntaxKind.OperatorDeclaration:
+                case SyntaxKind.ConversionOperatorDeclaration:
+                case SyntaxKind.AnonymousMethodExpression:
+                case SyntaxKind.SimpleLambdaExpression:
+                case SyntaxKind.ParenthesizedLambdaExpression:
+                    return (int)SyntaxKind.ReturnStatement;
 
-            var oldIfStatement = (IfStatementSyntax)ifStatementSyntax;
+                case SyntaxKind.DoStatement:
+                case SyntaxKind.WhileStatement:
+                case SyntaxKind.ForStatement:
+                case SyntaxKind.ForEachStatement:
+                case SyntaxKind.ForEachVariableStatement:
+                    return (int)SyntaxKind.ContinueStatement;
+            }
 
-            // For single line statement, we swap the TrailingTrivia to preserve the single line
-            StatementSyntax newIfNodeStatement = null;
-            ElseClauseSyntax newElseStatement = null;
+            return -1;
+        }
 
-            var hasNewLineAfterClosingBrace = oldIfStatement.Statement.GetTrailingTrivia().Any(trivia => trivia.Kind() == SyntaxKind.EndOfLineTrivia);
-            if (hasNewLineAfterClosingBrace)
+        protected override SyntaxNode GetJumpStatement(int rawKind)
+        {
+            switch ((SyntaxKind)rawKind)
             {
-                newIfNodeStatement = oldIfStatement.Else.Statement.Kind() != SyntaxKind.Block
-                    ? SyntaxFactory.Block(oldIfStatement.Else.Statement)
-                    : oldIfStatement.Else.Statement;
-                newElseStatement = oldIfStatement.Else.WithStatement(oldIfStatement.Statement);
+                case SyntaxKind.ContinueStatement:
+                    return SyntaxFactory.ContinueStatement();
+                case SyntaxKind.BreakStatement:
+                    return SyntaxFactory.BreakStatement();
+                case SyntaxKind.ReturnStatement:
+                    return SyntaxFactory.ReturnStatement();
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(rawKind);
             }
-            else
+        }
+
+        protected override StatementSyntax AsEmbeddedStatement(
+            StatementSyntax originalStatement,
+            IEnumerable<SyntaxNode> newStatements)
```

Comment:
is this not typeable as `IEnumerable<StatementSyntax>`?

Reply:
This is an override

Reply:
Even so, can the base abstract method be more strongly typed?